contract/src/airdrop/prepare.js

```diff
@@ -0,0 +1,267 @@
+import { M } from '@endo/patterns';
+import { makeDurableZone } from '@agoric/zone/durable.js';
+import { E } from '@endo/far';
+import { Far } from '@endo/marshal';
+import { AmountMath } from '@agoric/ertp';
+import { TimeMath, TimestampShape } from '@agoric/time';
+import '../../types.js'
+
+/**
+ * @typedef {object} EpochDetails
+ * @property {bigint} windowLength Length of epoch in seconds. This value is used by the contract's timerService to schedule a wake up that will fire once all of the seconds in an epoch have elapsed
```

Comment:
consider using `RelativeTime`.

```suggestion
 * @property {import('@agoric/time').RelativeTime} windowLength Length of epoch in seconds. This value is used by the contract's timerService to schedule a wake up that will fire once all of the seconds in an epoch have elapsed
```